### ISSUE TITLE
compileAndLint is 2.5 times slower than lint

### ISSUE THREAD

**Issue by BraisGabin** [arturbosch/detekt#2095]
I did a fast check and the line `KotlinScriptEngine.compile(content)` in `compileAndLint` slows down the tests a lot. In my computer the gradle task `:detekt-rules:test` takes ~82s with `compileAndLint` and ~32s with `lint` (2.5x faster).

Don't get me wrong: I like the safety that `compileAndLint` bring us. I move some test to `compileAndLint` and there was a lot of code that didn't compile. But this `compileAndLint` is not testing the production code. It's testing our tests.

For all that I said before, I think that we should run the line `KotlinScriptEngine.compile(content)` just in CI. This way we get the good of both parts. Fast tests in our end to test the production code and we ensure that the tests are correct in CI.

What do you think? This should be easy to accomplish with gradle.

**Comment by schalkms** [comment]
That's true.
We introduced this to migrate away from the case files in order to have tests that only check a single inline code snippet (#1089). Furthermore, we also wanted to have the Kotlin compiler as a safety net, since we saw that many code snippets were invalid Kotlin code.

You are right. It takes longer to have the Kotlin compiler compile the code snippets. It would be enough in my opinion to only run it once on Travis and once on AppVeyor.

It would be cool to have this improvement.
Do have an idea how to do this? @BraisGabin
